src/fsearch_window.c

```diff
@@ -628,6 +629,20 @@ on_listview_row_activated (GtkTreeView       *tree_view,
             BTreeNode * node = db_search_entry_get_node (entry);
             if (launch_node (node)) {
                 // open succeeded
+                fsearch_window_action_after_file_open(true);
+            } else {
+                // open failed
+                FsearchConfig *config = fsearch_application_get_config (FSEARCH_APPLICATION_DEFAULT);
+                if (config->show_dialog_failed_opening) {
```

Comment:
There's a bug here. This dialog should only be shown when `config->action_after_file_open_keyboard` or
`config->action_after_file_open_mouse` is `true`.